File: js/현수막(14716).js
const input = require('fs').readFileSync('/dev/stdin').toString().trim().split('\n');

const [M, N] = input[0].split(' ').map(Number);
graph = [];
for (let i=1; i <= M; i++) {
    graph.push(input[i].split(' ').map(Number));
}

const visited = Array.from({length: M}, () => Array(N).fill(false));
const dx = [-1,1,0,0,1,1,-1,-1]
const dy = [0,0,-1,1,1,-1,1,-1]

const bfs = (x,y) => {
    const queue = [];
    queue.push([x,y]);
    visited[x][y] = true;
    let index = 0

    while (index < queue.length) {
        const [x,y] = queue[index++];

        for (let i=0; i<8; i++) {
            const nx = x + dx[i];
            const ny = y + dy[i];

            if (nx >=0 && nx < M && ny >=0 && ny < N && !visited[nx][ny] && graph[nx][ny] === 1) {
                visited[nx][ny] = true;
                queue.push([nx,ny]);
            }
        }
    }
}

let cnt = 0;
for (let i=0; i<M; i++) {
    for (let j=0; j<N; j++) {
        if (graph[i][j] === 1 && !visited[i][j]) {
            bfs(i,j);
            cnt++;
        }
    }
}

console.log(cnt);

File: python/현수막(14716).py
from collections import deque

M, N = tuple(map(int, input().split()))

graph = [
    list(map(int, input().split()))
    for _ in range(M)
]

visited = [[False] * (N) for _ in range(M)]
dx = [-1,1,0,0,1,1,-1,-1]
dy = [0,0,-1,1,1,-1,1,-1]

def bfs(x,y):
    queue = deque()
    queue.append((x,y))
    visited[x][y] = True

    while queue:
        x,y = queue.popleft()

        for i in range(8):
            nx = x + dx[i]
            ny = y + dy[i]

            if nx >= 0 and nx < M and ny >= 0 and ny < N and visited[nx][ny] == False and graph[nx][ny] == 1:
                visited[nx][ny] = True
                queue.append((nx,ny))

cnt = 0 
for i in range(M):
    for j in range(N):
        if graph[i][j] == 1 and visited[i][j] == False:
            bfs(i,j)
            cnt += 1

print(cnt)